src/librustc_resolve/late/diagnostics.rs

```diff
@@ -168,9 +168,9 @@ impl<'a> LateResolutionVisitor<'a, '_, '_> {
         if ["this", "my"].contains(&&*item_str.as_str())
             && self.self_value_is_available(path[0].ident.span, span)
         {
-            err.span_suggestion(
+            err.span_suggestion_short(
                 span,
-                "did you mean",
+                "you might have meant to use `self` here instead",
```

Comment:
Is there a convention about this?
I remember some talks about the wording for "did you mean" and similar cases, but I don't remember seeing any conclusions written out anywhere.

Reply:
The conclusion was don't frame things as a question and make the message make sense mostly on its own. For a while now I've been changing "Did you mean" to more explicit messages. There aren't that many left anymore.